### PR TITLE
AlphaFold Mismatch Sequence

### PR DESCRIPTION
**Pull request by gustavo-salazar**
AlphaFold predictions are built using the sequences in UniProt. UniProt entries can be deleted or updated. For the former, we won't display the AlphaFold prediction... because we no longer show the UniProt protein. But in the latter, it might be an issue, especially if it is the sequence that has been updated.

This PR generates a warning when such a mismatch is detected, including a diff between the sequences.

Some proteins with mismatch for testing purposes:
```
A0A017SPL2, A0A023IAJ3, A0A075B784, A0A075B7D6, A0A075B7D9, A0A075B7E4, A0A075B7F2, A0A075B7H8, A0A075B7I5, A0A076FEI8, A0A087WNS9, A0A087WPT5, A0A087WPY4, A0A087WRY6, A0A087WVG0, A0A087WVG8, A0A087WVN8, A0A087WW20, A0A087WW56, A0A087WWF4, A0A087WWI6, A0A087WWP5, A0A087WWU1, A0A087WX00, A0A087WX35, A0A087WXI3, A0A087WXP2, A0A087WY29, A0A087WYC0, A0A087WYW5, A0A087WZ21, A0A087WZT3, A0A087X033, A0A087X070, A0A087X0C0, A0A087X0Y3, A0A087X183, A0A087X1J6, A0A087X1K7, A0A087X1R3, A0A087X1S4, A0A087ZK75, A0A089WV83, A0A089YLQ0
```

The message also appears in the AlphaFold subsection of entry pages, but I forgot which entry and protein combination I was using to test it 😬

Additionally, migrating to TS some of the AlphaFold components. I also started defining a better type for the Redux state, but still WIP.

### CODE REVIEW

**Review by typhainepl — APPROVED** [review]
Looking good to me :)

File: src/components/AlphaFold/Model/index.tsx [hunk 1]
```diff
@@ -118,11 +116,15 @@ const AlphaFoldModel = (
           </p>
         </>
       )}
+      <SequenceCheck
+        proteinAccession={proteinAcc}
+        alphaFoldSequence={data.payload?.[0]?.uniprotSequence}
+      />
```
Comment (matthiasblum) [inline src/components/AlphaFold/Model/index.tsx:122]: I would move the warning callout below the info callout informing users that the entry matches several proteins with structure predictions. The warning message should be as close as possible to the structure viewer.

**Review by matthiasblum — COMMENTED** [review]
Looks good. 

The only thing that I think needs changes is the "Show Diff" button which changes the width of the container. I guess it's because the `ReactDiffViewer` component prevents text wrapping. 

Personally, I'd simply remove the diff, and maybe display the number of mismatches instead (if it's easy to do, otherwise I'd just leave the warning message and that's it).

**Comment by gustavo-salazar** [comment]
@matthiasblum I like the diff widget, I think is quite informative, because it is a change of a single base, the model is still quite valid (e.g. `A0A017SPL2`) but if the change is bigger then is really no reason to trust the model (e.g. `A0A075B7F2`)

File: src/components/AlphaFold/Model/index.tsx
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply src/components/AlphaFold/Model/index.tsx:1]: Nope, the one that is currently used takes the protein accession from the props, and the othe from the state, which is importan for the case of multiple alphafold models in an entry page

**Pull request merged**